buildtools/bazel/bazel.go

```diff
@@ -1,16 +1,198 @@
+	lines := strings.Split(commandOutput, "\n")
+	if strings.Contains(lines[0], `version="1.1"`) {
+		lines[0] = `<?xml version="1.0" encoding="UTF-8"?>`
+	}
+
+	var query Query
+	err := xml.Unmarshal([]byte(strings.Join(lines, "\n")), &query)
+	if err != nil {
+		return nil, &errors.Error{
+			Cause:           err,
+			Troubleshooting: fmt.Sprintf("FOSSA was unable to read the xml output from the Bazel command. This may be related to conflicting versions of xml as version 1.1 is unsupported by the FOSSA CLI. If you run into this issue please send the results of the command `bazel query 'deps(%s)' --output xml` to fossa.", target),
+		}
+	}
+
+	packages := []pkg.ID{}
+	for _, rule := range query.Rules {
+		pack := pkg.ID{}
+		switch rule.Class {
+		case "go_library":
+			fallthrough
+		case "dbx_go_library":
+			pack.Type = pkg.Go
+			pack.Name = goNameFromRule(rule.Name, target)
+		case "dbx_npm_library":
+			pack.Type = pkg.NodeJS
+			requirement := valueFromKey(rule.Values, "npm_req")
+			pack.Name, pack.Revision = npmReqFromString(requirement)
+		case "py_library":
+		case "dbx_py_pypi_piplib":
+			pack.Type = pkg.Python
+			pack.Name = valueFromKey(rule.Values, "name")
+			pack.Revision = valueFromKey(rule.Values, "pip_version")
+		case "rust_library":
+			pack.Type = pkg.Rust
+			pack.Name = valueFromKey(rule.Values, "name")
+			pack.Revision = valueFromKey(rule.Values, "version")
+		case "cc_library":
+			fallthrough
+		case "dbx_thirdparty_cc_library":
+			splitLocation := strings.Split(rule.Location, ":")
+			if len(splitLocation) != 0 {
+				log.Debugf("cc library %s was unable to be uploaded because FOSSA was unable to find a directory for the library. rule: %+v", rule.Name, rule)
+				continue
+			}
+
+			trimmed := strings.TrimSuffix(strings.TrimSuffix(splitLocation[0], "BUILD"), "BUILD.bazel")
+			loc, err := fossa.UploadTarballDependency(trimmed, upload, true)
+			if err != nil {
+				log.Debugf("cc library %s was unable to be uploaded. rule: %+v. error: %s", rule.Name, rule, err.Error())
+				continue
+			}
+
+			pack.Type = pkg.Raw
+			pack.Name = loc.Project
+			pack.Revision = loc.Revision
+		default:
+			continue
+		}
+
+		packages = append(packages, pack)
+	}
+
+	return packages, nil
+}
+
+func goNameFromRule(ruleName string, target string) string {
+	trimmedTarget := strings.TrimSuffix(target, "...")
+	trimmedRuleName := strings.TrimPrefix(ruleName, trimmedTarget)
+	nameSplit := strings.Split(trimmedRuleName, ":")
+	if len(nameSplit) > 0 {
+		return nameSplit[0]
+	}
+	return ""
+}
+
+// Handle the npm_req field formats. Name and version are separated by "@" and "@" can be present in the name.
+func npmReqFromString(req string) (string, string) {
+	npmSpec := strings.Split(req, "@")
+
+	name, version := "", ""
+	switch len(npmSpec) {
+	case 3: // Handle requirements in the format "types/react-router-dom@1.0.0"
+		name = "@" + npmSpec[0] + npmSpec[1]
+		version = npmSpec[2]
+	case 2:
```

Comment:
if the package name contains "@" but the package doesn't have a version, e.g.,
`@types/blah`
this is going to return
```
name = ""
version="types/blah"
```